### PR TITLE
Adding Presto ip_prefix function

### PR DESCRIPTION
**Pull request by bikash-c**

### CODE REVIEW

File: velox/functions/prestosql/IPFunctions.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#include "velox/functions/lib/string/StringImpl.h"
+
+namespace facebook::velox::functions {
+
+template <typename T>
+struct IPPrefixFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+  FOLLY_ALWAYS_INLINE bool call(out_type<Varchar>& result,
+                                const arg_type<Varchar>& ip ,
+                                const uint8_t prefixBits) {
```
Comment (kgpai) [inline velox/functions/prestosql/IPFunctions.h:30]: I dont believe this is equivalent to what presto has - would we need to cast to tinyint to call this function ? Presto has this as bigint .

File: velox/functions/prestosql/IPFunctions.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    const folly::IPAddress ipAddress = folly::IPAddress(ip);
+    auto ipResult = ipAddress.mask(prefixBits);
+    result.setNoCopy(StringView(ipResult.str()));
```
Comment (kgpai) [inline velox/functions/prestosql/IPFunctions.h:34]: Unfortunately this code might result in segfaults. 
The way to do this, would be to set result.reserve(strlen(ipResult.str())), then copy over the string to the address pointed over by result.data().

File: velox/functions/prestosql/tests/IPFunctionsTest.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#include "folly/IPAddress.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::functions::test;
+
+namespace {
+class IPFunctionsTest : public FunctionBaseTest {
+ protected:
+
+  std::optional<std::string> getIPPrefix(const std::optional<std::string> input, std::optional<int> mask) {
```
Comment (kgpai) [inline velox/functions/prestosql/tests/IPFunctionsTest.cpp:28]: any reason why input and mask optional ?

File: velox/functions/prestosql/tests/IPFunctionsTest.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    auto result = evaluateOnce<std::string>("ip_prefix(c0, c1)", input, mask);
+    return result;
```
Comment (kgpai) [inline velox/functions/prestosql/tests/IPFunctionsTest.cpp:30]: nit: return evaluateOnce<std::string>....

File: velox/functions/prestosql/tests/IPFunctionsTest.cpp (hunk 3 above)
Comment (kgpai) [reply velox/functions/prestosql/tests/IPFunctionsTest.cpp:28]: Also if your function take uint8_t , why pass int here ?

File: velox/functions/prestosql/tests/IPFunctionsTest.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  }
+};
+
+TEST_F(IPFunctionsTest, IPv4) {
+
+  EXPECT_EQ("10.135.23.0", getIPPrefix("10.135.23.12",8));
+  EXPECT_EQ("10.132.0.0", getIPPrefix("10.135.23.12",18));
+  EXPECT_EQ("0.0.0.0", getIPPrefix("10.135.23.12",32));
+  EXPECT_EQ("10.135.23.12", getIPPrefix("10.135.23.12",0));
```
Comment (kgpai) [inline velox/functions/prestosql/tests/IPFunctionsTest.cpp:39]: Can you add tests for null strings, non ip strings etc..

File: velox/docs/functions/ip.rst [hunk 6]
```diff
@@ -0,0 +1,9 @@
+=============
+IP Functions
+=============
+
+IP mask function
+
+.. function:: ip_prefix(ip_address, prefix_bits) → varchar
+
+Returns the IP prefix of a given ip_address with subnet size of prefix_bits. ip_address will be of type VARCHAR.
```
Comment (kgpai) [inline velox/docs/functions/ip.rst:9]: nit highlight ip_address like `ip_address`, similarly `prefix_bits`, etc..

**Comment by mbasmanova** [comment]
Closing stale PRs. Feel free to re-open if decide to continue working on this.

**Pull request closed without merging**